x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/query/QueryVector.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.vectors.query;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Provide consistent access to query vector elements from a backing list or float array.  Used to provide
+ * a flexible API for DenseVector in painless, which does not have method overloading.
+ */
+public class QueryVector {
```

Comment:
Is there still a benefit to introducing this wrapper class, or could we just use `List<Number>` directly in the vector functions? (Sorry if I missed this in our conversations)

Reply:
The benefit is avoiding an unchecked cast and a nicer error message if the user sends in a mixed type `List`.

We could avoid the wrapper and use a helper instead method, that would still give a nice error and remove the `QueryVector` helper.

Reply:
Removed `QueryVector`.